packages/web-components/src/anchor-button/anchor-button.ts

```diff
@@ -230,6 +231,18 @@ export class AnchorButton extends BaseAnchor {
   @attr
   public appearance?: AnchorButtonAppearance | undefined;
 
+  /**
+   * Handles changes to appearance attribute custom states
+   * @param prev - the previous state
+   * @param next - the next state
+   */
+  public appearanceChanged(prev: AnchorButtonAppearance | undefined, next: AnchorButtonAppearance | undefined) {
+    if (prev) {
+      toggleState(this.elementInternals, prev, false);
+    }
+    toggleState(this.elementInternals, `${next}`, true);
+  }
```

Comment:
I think long-term we can remove the duplication here across appearances, shape, size, etc in many places, but for now I want to make the state changes before we jump to an abstraction.